### PR TITLE
Enable and Allow the customization of serviceAccountName

### PR DESCRIPTION
**Pull request by falcon-pioupiou**
- [Fix] the hardcoded serviceAccountName in the daemonset
- [Fix] the hardcoded serviceAccountName in the clusterrolebinding
- [Fix] the hardcoded serviceAccountName in the SecurityContextConstraints
- [Fix] the hardcoded serviceAccountName in the ServiceAccount
- [Update] the values schema to support the serviceAccountName at root of values
- [Update] the ServiceAccount creation based on values (default : true as the account was created before but not used)
- [Add] the usage of the serviceAccountName for the injector
- [Add] Values : add the custom value for the serviceAccountName

Those changes allows the Falcon Container to be deployed on GKE cluster with Workload Identity (config actually recommended and not yet enforced on GCP side)
https://cloud.google.com/iam/docs/best-practices-for-using-workload-identity-federation

it should also support AWS OIDC as a K8S serviceAccount with annotations (for role binding) is needed

### CODE REVIEW

File: helm-charts/falcon-sensor/values.yaml [hunk 1]
```diff
@@ -84,13 +84,17 @@ container:
...
-  #   memory: 128Mi
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
     requests:
       cpu: 10m
       memory: 20Mi
 
+serviceAccount:
+  create: true
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/values.yaml:95]: This isn't needed as the service accounts should always be there.

File: helm-charts/falcon-sensor/templates/daemonset.yaml [hunk 2]
```diff
@@ -133,7 +133,7 @@ spec:
         - name: falconstore-dir
           hostPath:
             path: /var/lib/crowdstrike
-      serviceAccountName: crowdstrike-falcon-sa
+      serviceAccountName: {{ include "falcon-sensor.serviceAccountName" .}}
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/templates/daemonset.yaml:136]: This should be `{{ .Values.serviceAccount }}`

File: helm-charts/falcon-sensor/templates/container_deployment_webhook.yaml [hunk 3]
```diff
@@ -93,6 +93,7 @@ spec:
       - name: {{ include "falcon-sensor.name" . }}-tls-certs
         secret:
           secretName: {{ include "falcon-sensor.name" . }}-tls
+      serviceAccountName: {{ include "falcon-sensor.serviceAccountName" . }}
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/templates/container_deployment_webhook.yaml:96]: This should be `{{ .Values.serviceAccount }}`

File: helm-charts/falcon-sensor/templates/serviceaccount.yaml [hunk 4]
```diff
@@ -1,12 +1,14 @@
+{{- if .Values.serviceAccount.create }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: crowdstrike-falcon-sa
+  name: {{ include "falcon-sensor.serviceAccountName" . }}
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/templates/serviceaccount.yaml:5]: This should be `{{ .Values.serviceAccount }}`

File: helm-charts/falcon-sensor/templates/serviceaccount.yaml [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
   namespace: {{ .Release.Namespace }}
   labels:
-    app: {{ include "falcon-sensor.name" . }}
-    app.kubernetes.io/name: {{ include "falcon-sensor.name" . }}
+    app: {{ include "falcon-sensor.name" .}}
+    app.kubernetes.io/name: {{include "falcon-sensor.name" . }}
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/templates/serviceaccount.yaml:9]: Please undo these changes

File: helm-charts/falcon-sensor/templates/serviceaccount.yaml [hunk 6]
```diff
@@ -1,12 +1,14 @@
+{{- if .Values.serviceAccount.create }}
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/templates/serviceaccount.yaml:1]: This isn't needed

File: helm-charts/falcon-sensor/values.yaml [hunk 7]
```diff
@@ -84,13 +84,17 @@ container:
...
-  #   cpu: 100m
-  #   memory: 128Mi
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
     requests:
       cpu: 10m
       memory: 20Mi
 
+serviceAccount:
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/values.yaml:94]:
```suggestion
serviceAccount: crowdstrike-falcon-sa
```

File: helm-charts/falcon-sensor/values.yaml [hunk 8]
```diff
@@ -84,13 +84,17 @@ container:
...
+  create: true
+  name: crowdstrike-falcon-sa
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/values.yaml:96]: This can be merged like:

```
serviceaccount: serviceAccount
```

File: helm-charts/falcon-sensor/templates/node_scc.yaml [hunk 9]
```diff
@@ -31,7 +31,7 @@ requiredDropCapabilities:
 defaultAddCapabilities:
 allowedCapabilities:
 users:
-- crowdstrike-falcon-sa
+- {{include "falcon-sensor.serviceAccountName" .}}
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/templates/node_scc.yaml:34]: This should be `{{ .Values.serviceAccount }}`

File: helm-charts/falcon-sensor/templates/clusterrolebinding.yaml [hunk 10]
```diff
@@ -17,7 +17,7 @@ subjects:
   name: system:authenticated
 {{- end }}
 - kind: ServiceAccount
-  name: crowdstrike-falcon-sa
+  name: {{include "falcon-sensor.serviceAccountName" .}}
```
Comment (redhatrises) [inline helm-charts/falcon-sensor/templates/clusterrolebinding.yaml:20]: This should be `{{ .Values.serviceAccount }}`

**Review by redhatrises — CHANGES_REQUESTED** [review]
Thanks for submitting this PR! There are a couple of minor changes that need to be made.

File: helm-charts/falcon-sensor/templates/daemonset.yaml (hunk 2 above)
Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/daemonset.yaml:136]: I'm changing to `{{ .Values.serviceAccount }}` but I was using the serviceAccountName helper, accessing directly to the value we can clean up the code by removing the serviceAccountName Helper in `_helpers.tpl` (I will suggest this modification as well)

Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/daemonset.yaml:136]: definition : 
- `helm-charts/falcon-sensor/templates/_helpers.tpl:56:{{- define "falcon-sensor.serviceAccountName" -}}`

places where this helper is used: 
- `helm-charts/falcon-sensor/templates/daemonset.yaml:136:      serviceAccountName: {{ include "falcon-sensor.serviceAccountName" .}}`
- `helm-charts/falcon-sensor/templates/node_scc.yaml:34:- {{include "falcon-sensor.serviceAccountName" .}}`
- `helm-charts/falcon-sensor/templates/container_deployment_webhook.yaml:96:      serviceAccountName: {{ include "falcon-sensor.serviceAccountName" . }}`
- `helm-charts/falcon-sensor/templates/serviceaccount.yaml:4:  name: {{ include "falcon-sensor.serviceAccountName" . }}`
- `helm-charts/falcon-sensor/templates/clusterrolebinding.yaml:20:  name: {{include "falcon-sensor.serviceAccountName" .}}`

Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/daemonset.yaml]: I have cleaned up everything to match `{{ .Values.serviceAccount }}`

File: helm-charts/falcon-sensor/templates/serviceaccount.yaml (hunk 4 above)
Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/serviceaccount.yaml]: done

File: helm-charts/falcon-sensor/templates/serviceaccount.yaml (hunk 6 above)
Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/serviceaccount.yaml]: done

File: helm-charts/falcon-sensor/templates/node_scc.yaml (hunk 9 above)
Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/node_scc.yaml]: done

File: helm-charts/falcon-sensor/templates/clusterrolebinding.yaml (hunk 10 above)
Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/clusterrolebinding.yaml]: done

File: helm-charts/falcon-sensor/templates/serviceaccount.yaml (hunk 5 above)
Comment (falcon-pioupiou) [reply helm-charts/falcon-sensor/templates/serviceaccount.yaml]: done
